packages/ui-core/src/components/typography-settings/typography-settings.component.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { FC, RefObject } from 'react';
+import { Icon } from '@keen.io/icons';
+import { colors } from '@keen.io/colors';
+import { Container, Separator, TextOption } from './typography-settings.styles';
+import { FontSize, TextAlignment } from './components';
+import { DEFAULT_FONT_SIZES, FONT_STYLES } from './constants';
+import { FontSettings } from './types';
+import { Color } from '../index';
+
+type Props = {
+  /** Settings that are customizable */
+  settings: FontSettings;
+  /** Function that will be called on settings change with new settings as a parameter */
+  onChange: (settings: FontSettings) => void;
+  /** Color suggestions that will be available in color selector */
+  colorSuggestions?: string[];
+  /** Font size suggestions that will be available in font size dropdown */
+  fontSizeSuggestions?: number[];
+  /** Ref to scrollable parent element - it can be used to hide color picker or dropdowns on scroll event */
+  scrollableContainerRef?: RefObject<HTMLDivElement>;
+};
+
+const TypographySettings: FC<Props> = ({
+  settings,
+  onChange,
```

Comment:
As we discussed - could You please add possibility to specify which properties are allowed to edit like for eg. text alignment etc.